File: api/src/main/java/dev/ulloasp/mlsuite/organization/application/dto/OrganizationMembershipDto.java
package dev.ulloasp.mlsuite.organization.application.dto;

import java.time.OffsetDateTime;

import dev.ulloasp.mlsuite.organization.domain.model.OrganizationMembership;

public record OrganizationMembershipDto(
        Long id,
        Long organizationId,
        Long userId,
        String fullName,
        String email,
        String avatarUrl,
        String role,
        String status,
        OffsetDateTime createdAt) {

    public static OrganizationMembershipDto from(OrganizationMembership membership) {
        return new OrganizationMembershipDto(
                membership.getId(),
                membership.getOrganization().getId(),
                membership.getUser().getId(),
                membership.getUser().getFullName(),
                membership.getUser().getEmail(),
                membership.getUser().getAvatarUrl(),
                membership.getRole().name(),
                membership.getStatus().name(),
                membership.getCreatedAt());
    }
}

File: frontend/src/api/workspace/dtos/organization-membership-dto.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { OrganizationRole, MembershipStatus } from "./index";

export interface OrganizationMembershipDto {
  id: number;
  organizationId: number;
  userId: number;
  fullName: string;
  email: string;
  avatarUrl?: string | null;
  role: OrganizationRole;
  status: MembershipStatus;
  createdAt: string;
}
